File: Arrays/palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
	    for i in s:
		    if s[::-1]!=s[::1]:
		       return False
		    else:
		        return True
s=eval(input("Enter array size"))
arr=[]
for i in range(s):
    arr.append(eval(input("Enter value")))
sol=Solution()
a=sol.isPalindrome(arr)
print(a)

File: Arrays/palindrome.c
#include <stdio.h>
#include <stdbool.h>

bool isPalindrome(int arr[], int size) {
    for (int i = 0; i < size / 2; i++) {
        if (arr[i] != arr[size - 1 - i]) {
            return false;
        }
    }
    return true;
}

int main() {
    int arr[100], size;

    printf("Enter array size: ");
    scanf("%d", &size);

    printf("Enter %d elements:\n", size);
    for (int i = 0; i < size; i++) {
        printf("Element %d: ", i + 1);
        scanf("%d", &arr[i]);
    }

    if (isPalindrome(arr, size)) {
        printf("Palindrome ?\n");
    } else {
        printf("Not a Palindrome ?\n");
    }

    return 0;
}
